tool/planet/cfg.go

```diff
@@ -154,12 +156,42 @@ func (cfg *Config) checkAndSetDefaults() (err error) {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	if err := verifyPodSubnetSize(cfg.PodSubnetSize, cfg.PodCIDR); err != nil {
+		return trace.Wrap(err, "failed to verify pod subnet size")
+	}
+
 	if cfg.VxlanPort <= 0 {
 		cfg.VxlanPort = DefaultVxlanPort
 	}
 	return nil
 }
 
+// verifyPodSubnetSize verifies that the subnet size is not too small and verifies
+// that the subnet size is not larger than the network CIDR range.
+func verifyPodSubnetSize(subnetSize int, cidr kv.CIDR) error {
+	// The minimum subnet size accepted by flannel is /28:
+	// https://github.com/gravitational/flannel/blob/master/subnet/config.go#L70-L74
+	if subnetSize > 28 {
+		return trace.BadParameter("pod subnet is too small. Minimum useful network prefix is /28: %d", subnetSize)
```

Comment:
Nit: It's a bit inconsistent that here the `subnetSize` is formatted into the message and below it's attached as a field. I would pick one way and stick to it.